pkg/util/system/cgroup_resource.go

```diff
@@ -18,124 +18,290 @@ package system
-	MemUsePriorityOom   = CgroupFile{ResourceFileName: MemUsePriorityOomFileName, Subfs: CgroupMemDir, IsAnolisOS: true, Validator: MemUsePriorityOomValidator}
-	MemOomGroup         = CgroupFile{ResourceFileName: MemOomGroupFileName, Subfs: CgroupMemDir, IsAnolisOS: true, Validator: MemOomGroupValidator}
-	MemWmarkMinAdj      = CgroupFile{ResourceFileName: MemWmarkMinAdjFileName, Subfs: CgroupMemDir, IsAnolisOS: true, Validator: MemWmarkMinAdjValidator}
-	MemWmarkScaleFactor = CgroupFile{ResourceFileName: MemWmarkScaleFactorFileName, Subfs: CgroupMemDir, IsAnolisOS: true, Validator: MemWmarkScaleFactorFileNameValidator}
-	MemMin              = CgroupFile{ResourceFileName: MemMinFileName, Subfs: CgroupMemDir, IsAnolisOS: true, Validator: MemMinValidator}
-	MemLow              = CgroupFile{ResourceFileName: MemLowFileName, Subfs: CgroupMemDir, IsAnolisOS: true, Validator: MemLowValidator}
-	MemHigh             = CgroupFile{ResourceFileName: MemHighFileName, Subfs: CgroupMemDir, IsAnolisOS: true, Validator: MemHighValidator}
-
-	BlkioReadIops  = CgroupFile{ResourceFileName: BlkioTRIopsFileName, Subfs: CgroupBlkioDir, IsAnolisOS: false, Validator: BlkioReadIopsValidator}
-	BlkioReadBps   = CgroupFile{ResourceFileName: BlkioTRBpsFileName, Subfs: CgroupBlkioDir, IsAnolisOS: false, Validator: BlkioReadBpsValidator}
-	BlkioWriteIops = CgroupFile{ResourceFileName: BlkioTWIopsFileName, Subfs: CgroupBlkioDir, IsAnolisOS: false, Validator: BlkioWriteIopsValidator}
-	BlkioWriteBps  = CgroupFile{ResourceFileName: BlkioTWBpsFileName, Subfs: CgroupBlkioDir, IsAnolisOS: false, Validator: BlkioWriteBpsValidator}
-
-	CPUProcs = CgroupFile{ResourceFileName: ProcsFileName, Subfs: CgroupCPUDir, IsAnolisOS: false}
+	DefaultFactory = NewCgroupResourceFactory()
+
+	CPUStat      = DefaultFactory.New(CPUStatName, CgroupCPUDir)
+	CPUShares    = DefaultFactory.New(CPUSharesName, CgroupCPUDir)
+	CPUCFSQuota  = DefaultFactory.New(CPUCFSQuotaName, CgroupCPUDir)
+	CPUCFSPeriod = DefaultFactory.New(CPUCFSPeriodName, CgroupCPUDir)
+	CPUBurst     = DefaultFactory.New(CPUBurstName, CgroupCPUDir).WithValidator(CPUBurstValidator).WithCheckSupported(SupportedIfAnolisOS)
+	CPUBVTWarpNs = DefaultFactory.New(CPUBVTWarpNsName, CgroupCPUDir).WithValidator(CPUBvtWarpNsValidator).WithCheckSupported(SupportedIfAnolisOS)
+	CPUTasks     = DefaultFactory.New(CPUTasksName, CgroupCPUDir)
+	CPUProcs     = DefaultFactory.New(CPUProcsName, CgroupCPUDir)
+
+	CPUSet = DefaultFactory.New(CPUSetCPUSName, CgroupCPUSetDir)
+
+	CPUAcctUsage = DefaultFactory.New(CPUAcctUsageName, CgroupCPUAcctDir)
+
+	MemoryLimit            = DefaultFactory.New(MemoryLimitName, CgroupMemDir)
+	MemoryUsage            = DefaultFactory.New(MemoryUsageName, CgroupMemDir)
+	MemoryStat             = DefaultFactory.New(MemoryStatName, CgroupMemDir)
+	MemoryWmarkRatio       = DefaultFactory.New(MemoryWmarkRatioName, CgroupMemDir).WithValidator(MemoryWmarkRatioValidator).WithCheckSupported(SupportedIfAnolisOS)
```

Comment:
OK, we can check inside the `IfAnolisOS()`

Reply:
I mean we should not use `IfAnolisOS` to check wether a cgroup resource is supported. Because some versions of AnolisOS may only support few cgroup resource(e.g. MemoryWmarkRatio supported but bvt is not supported)
for resources like MemoryWmarkRatio, we should use `if-the-file-exist` for `WithCheckSupported`.
for resources like `bvt`, we should use more complex function.

Reply:
fixed. check if file exist in `IsSupported()`